### PR TITLE
DEVPROD-2450 Redirect host pages to spruce if spruce enabled

### PR DESCRIPTION
**Pull request by bynn**
DEVPROD-2450

### Description
Host page will auto redirect to spruce just like spawn page does 

### Testing
Tested that host pages will auto redirect to spruce on staging
ie
https://evergreen-staging.corp.mongodb.com/host/i-00ab599007f7dd911 redirects to  https://spruce-staging.corp.mongodb.com/host/i-00ab599007f7dd911

### CODE REVIEW

File: service/middleware.go [hunk 1]
```diff
@@ -89,6 +90,15 @@ func RedirectSpruceUsers(w http.ResponseWriter, r *http.Request, redirect string
 	return true
 }
 
+// RedirectIfSpruceSet redirects the user to spruce only if they aren't visiting this page from spruce already and have spruce enabled
```
Comment (malikchaya2) [inline service/middleware.go:93]: nit: do they have to be on spruce already? Based on your test it looks like this works from evergreen too.

**Review by malikchaya2 — APPROVED** [review]
approved with nit

File: service/middleware.go (hunk 1 above)
Comment (bynn) [reply service/middleware.go:93]: I didn't write this comment but it's saying that the function does not do anything if you're already clicking into a spruce page ie clicking https://spruce-staging.corp.mongodb.com/host/i-00ab599007f7dd911 will do nothing

Comment (malikchaya2) [reply service/middleware.go:93]: oh haha I read "aren't" as "are"
thanks!

**Pull request merged**